simulator/scheduler/scheduler.go

```diff
@@ -122,18 +122,32 @@ func (s *Service) StartScheduler(versionedcfg *configv1.KubeSchedulerConfigurati
 		return xerrors.Errorf("New extender service: %w", err)
 	}
 
-	cfg, err := convertConfigurationForSimulator(versionedcfg, s.simulatorPort)
+	// Override the Extenders config so that the connection is directed to the simulator server.
+	extender.OverrideExtendersCfgToSimulator(versionedcfg, s.simulatorPort)
+
+	versioned, err := ConvertConfigurationForSimulator(versionedcfg)
 	if err != nil {
 		return xerrors.Errorf("convert scheduler config to apply: %w", err)
 	}
-	registry, err := plugin.NewRegistry(s.sharedStore, cfg)
+
+	cfg, err := ConvertSchedulerConfigToInternalConfig(versioned)
+	if err != nil {
+		return xerrors.Errorf("convert scheduler config to internal one: %w", err)
+	}
+
+	cfg, err = filterOutNonAllowedChangesOnCfg(cfg)
+	if err != nil {
+		return xerrors.Errorf("filter out non allowed changes: %w", err)
+	}
+
```

Comment:
no change in behavior. 
Just changing the parameters on each funcs, and moves logic from one to another.